homeassistant/components/recorder/purge.py

```diff
@@ -101,3 +59,67 @@ def purge_old_data(instance, purge_days: int, repack: bool) -> bool:
     except SQLAlchemyError as err:
         _LOGGER.warning("Error purging history: %s", err)
     return True
+
+
+def _select_event_ids_to_purge(session, purge_before):
+    """Return a list of event ids to purge."""
+    events = (
+        session.query(Events.event_id)
+        .filter(Events.time_fired < purge_before)
+        .limit(MAX_ROWS_TO_PURGE)
+        .all()
+    )
+    _LOGGER.debug("Selected %s event ids to remove", len(events))
+    return [event.event_id for event in events]
+
+
+def _select_state_ids_to_purge(session, event_ids):
+    """Return a list of state ids to purge."""
+    states = session.query(States.state_id).filter(States.event_id.in_(event_ids)).all()
```

Comment:
```suggestion
    if not event_ids:
        return []
    states = session.query(States.state_id).filter(States.event_id.in_(event_ids)).all()
```
Maybe return an empty list if the `event_ids` list is empty?